source/manual/test-and-build-a-project-with-github-actions.html.md

```diff
@@ -0,0 +1,266 @@
+  triggering [publishing end-to-end tests][e2e-tests] or triggering a
+  deployment)
+- Jenkins can be used for any services that require authentication to external
+  systems (for example, [releasing a gem
+  automatically][gem-release] during a master build), since secrets are not
+  permitted within GitHub Actions
+- As GitHub Actions are not constrained by the [GOV.UK Puppet
+  configuration][govuk-puppet] a wider range of software is available (for
+  example, Node.js versions past 8)
+- As GitHub Actions are not executed within GDS internal network they are a safer
+  choice for testing contributions from external partners (such as pull
+  requests from members of the public)
+
+## GOV.UK Conventions for GitHub Actions
+
+### Name of CI workflow file
+
+You should name the file used for configuring the CI workflow `ci.yml` and name
+the main task "Continuous Integration".
+
+### When the CI workflow should run
+
+The workflow should be configured to run on [`push`][push-event] and
+[`pull_request`][pull-request-event] events. The `push` event occurs for every
+push to the repository and this will generate a build against the branch
+that was pushed. The `pull_request` event occurs whenever a pull request
+is opened and it is triggered again any time the pull request changes (such
+as extra commits or rebasing), these differ to the `push` event by merging in
+the master branch before running operations. Pull requests from forks of a
+repo will only trigger the `pull_request` event.
+
+For example:
+
+```yml
+# ./github/workflows/ci.yml
+name: Continuous Integration
+on: [push, pull_request]
+```
+
+### Prefer a single or small number of steps within a CI workflow
+
+Rather than having a number of granular steps (such as: "Lint Ruby", "Lint JS",
+"Run Ruby tests", "Run JS Tests") prefer a single step (such as `bundle exec
+rake`) where more of the CI task configuration is handled in application code.
+
+This is so we can:
+
+- keep CI configurations simple and consistent;
+- make it easier to replicate CI checks in a development environment.
+
+### Base your workflow on one of our documented examples
+
+GitHub Actions currently [lack templating functionality][action-templates]
+to enable re-use of workflow configuration across projects. Therefore, in order
+to maintain consistency across GOV.UK projects, it is beneficial that we
+document and re-use common practices.
+
+This documentation contains examples of a number of common build needs for
+GOV.UK projects. You are encouraged to base your projects configuration on
+these and to follow their conventions. If your preferences, or common action
+conventions, deviate from these examples you are encouraged to open a PR to
+this repository to propose a change here. This is preferential to taking a
+project in a different direction and losing consistency.
+
+## GitHub Action Examples
+
+### Simple Ruby Application
+
+```yml
+# .github/workflows/ci.yml
+name: Continuous Integration
```

Comment:
Could you help me understand the advantage here? I suspect this will override the current GitHub UI, which prints the path to the file, which is useful if you're new.

Reply:
It will indeed. I was wondering about this too.

Reply:
I put some text about this in the PR description. Myself, I don't mind if we give workflows a name or not but would prefer it to be consistent - given we have some workflows with names and some without.

The UI does show the name, the path is the fallback, though clicking on any action shows you the workflow file so I don't think it obscures much.

My logic was that it was a more positive approach to suggest a consistent name than to say "do not set a name" and that it meant that any potential future actions also didn't need to be no name for consistency too. 

Where do you two stand on it? What would be your preference for guidance?

Reply:
For me, the less text we have, the more consistent we're likely to be (plus less typing). In this case, I don't find the name particularly helpful. As long as we don't need to set a name on any future actions, then it seems OK to not have one ever - this might encourage people to use a descriptive name for the file, and avoids it getting out-of-sync with any name.

Reply:
Thinking more about it, like Ben I'd be in favour of `.github/workflows/ci.yml` being the filename and each action having no defined `name`.

Reply:
Sure - that's cool. We can go with no name.